src/main/java/com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java

```diff
@@ -325,11 +324,8 @@ private void checkExistenceOfRequiredTranslations(ResourceBundle bundle) {
      * @param fileName file name.
      */
     private void logMissingTranslation(String filePath, String fileName) {
-        final MessageDispatcher dispatcher = getMessageDispatcher();
-        dispatcher.fireFileStarted(filePath);
```

Comment:
@Nimfadora `TranslationCheck` is our one and only multi-file validation check. So it is very "special".

I believe it does need these fires of started and finished.
Please can confirm by running CLI using `-f xml` for XML output. The reason why this option is important as `fireFileStarted` and such control the XML layout and before recent changes, specifically printed out the file tags for the output.

You can see the code https://github.com/checkstyle/checkstyle/blob/07a730c0d1b89593e17fa8920faf9c226200e7b5/src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java#L131
and https://github.com/checkstyle/checkstyle/blob/07a730c0d1b89593e17fa8920faf9c226200e7b5/src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java#L140

Reply:
Here are three outputs, without all (`fireFileStarted`, `fireFileFinished`), with only (`fireFileFinished`) and with both (`fireFileStarted` & `fireFileFinished`). 
Second and third are equal. First differs from them like this:
1) it doesn't have new line in the end of the file
2) it doesn't have duplicate line with base property file

[translation_only_finished.txt](https://github.com/checkstyle/checkstyle/files/1272137/translation_only_finished.txt)
[translation_started_finished.txt](https://github.com/checkstyle/checkstyle/files/1272139/translation_started_finished.txt)
[translation_without_all.txt](https://github.com/checkstyle/checkstyle/files/1272138/translation_without_all.txt)

If we don't care on duplicate (or maybe it is the right way of displaying this), I will leave only `fireFileFinished`. `fireFileStarted` only creates empty `FileMessages`, but in other methods of the class the `putIfAbsent` is used, so there is no need of creation empty `FileMessages`.

Reply:
Ok, I was expecting either an `<error>` tags without a corresponding `<file>` tag or no errors at all, but after looking deeper I see why this doesn't happen and it runs correctly.

After the check `destroy` executes from the violation, we run `fireAuditFinished`
https://github.com/checkstyle/checkstyle/blob/07a730c0d1b89593e17fa8920faf9c226200e7b5/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java#L219-L223
and the XML logger clears out it's file cache if it hadn't before.
https://github.com/checkstyle/checkstyle/blob/07a730c0d1b89593e17fa8920faf9c226200e7b5/src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java#L117-L118

So correct output is still here only because XML logger was extra careful in clearing out it's cache.
@romani Should this behavior stay. It is weird we fire errors without a start/end file, but this check is 'special' as it already is. I assume removing the `fileMessages.forEach(this::writeFileMessages);` would solve the pitest issue in this check.

> or maybe it is the right way of displaying this

We shouldn't display duplicates, but this is going to be the limitation of multi-file validation. Multi-file modules require to print violations after all other modules have executed. Unless, we keep all violations in memory, we will possibly have duplicate file tags between multi-file modules and normal modules.

Reply:
hmm.. for some reason gihub lost my comment for this topic....

Summary:
We will no do any changes related to multi-file support now, it is out of context.
@rnveach , please create separate issue for this.

Our focus is to enforce pitest over code, so we need to focus on coverage with minimal changes to checks or other checkstyle.

Reply:
Issue is at #5057 .